src/operator/correlation-inl.h

```diff
@@ -232,10 +232,10 @@ void Init(const std::vector<std::pair<std::string, std::string> >& kwargs) overr
                  std::vector<int> *out_type,
                  std::vector<int> *aux_type) const override {
     int dtype = (*in_type)[0];
-    type_assign(&(*in_type)[1], dtype);
-    type_assign(&(*out_type)[0], dtype);
-    type_assign(&(*out_type)[1], dtype);
-    type_assign(&(*out_type)[2], dtype);
+    type_assign(&dtype, (*in_type)[1]);
```

Comment:
I think you can just call `ElemwiseType<2, 3>(attrs, in_attrs, out_attrs)` to achieve the same purpose.

Reply:
This function does not have attrs argument, I guess what we have here is the only way to go.

Reply:
is it just trying to set all of the inputs to the first non -1 output type?

Reply:
Yes, all input and output types to the inferred type.

Reply:
sorry your statement there is kind of vague. can you please explain what you’re trying to do here? pretend i am 5 years old...

Reply:
also, you can just pass a blank NodAttrs if necessary

Reply:
also, you can just pass a blank NodAttrs if necessary